sdk/nodejs/runtime/closure/createClosure.ts

```diff
@@ -1158,34 +1176,112 @@ function getOrCreateEntry(
+        }
+
+        const envEntries = [...emptyOutputEntry.object.env.entries()];
+        if (envEntries.length !== 1) {
+            throw new Error("Expected SerializedOutput object to only have one property: " + envEntries.length);
+        }
+
+        const [envEntry] = envEntries[0];
+        if (envEntry.json !== "value") {
+            throw new Error("Expected SerializedOutput object sole property to be called 'value': " + envEntry.json);
+        }
+
+        // Everything looked good.  Replace the `"value" -> undefined-Entry` mapping in this entry
+        // with `"value" -> V-Entry`
+        emptyOutputEntry.object.env.set(envEntry,  {entry: valEntry});
+        return emptyOutputEntry;
+    }
+}
+
+async function isOutputAsync(obj: any): Promise<boolean> {
+    return resource.Output.isInstance(obj);
 }
 
 // Is this a constructor derived from a noCapture constructor.  if so, we don't want to
 // emit it.  We would be unable to actually hook up the "super()" call as one of the base
 // constructors was set to not be captured.
-function isDerivedNoCaptureConstructor(func: Function) {
+async function isDerivedNoCaptureConstructorAsync(func: Function): Promise<boolean> {
     for (let current: any = func; current; current = Object.getPrototypeOf(current)) {
-        if (current && current.doNotCapture) {
+        if (await hasTruthyMemberAsync(current, "doNotCapture")) {
             return true;
         }
     }
 
     return false;
 }
 
-// These modules are built-in to Node.js, and are available via `require(...)`
-// but are not stored in the `require.cache`.  They are guaranteed to be
-// available at the unqualified names listed below. _Note_: This list is derived
-// based on Node.js 6.x tree at: https://github.com/nodejs/node/tree/v6.x/lib
-const builtInModuleNames = [
-    "assert", "buffer", "child_process", "cluster", "console", "constants", "crypto",
-    "dgram", "dns", "domain", "events", "fs", "http", "https", "module", "net", "os",
-    "path", "process", "punycode", "querystring", "readline", "repl", "stream", "string_decoder",
-    /* "sys" deprecated ,*/ "timers", "tls", "tty", "url", "util", "v8", "vm", "zlib",
-];
-const builtInModules = new Map<any, string>();
-for (const name of builtInModuleNames) {
-    builtInModules.set(require(name), name);
+let builtInModules: Promise<Map<any, string>> | undefined;
+function getBuiltInModules(): Promise<Map<any, string>> {
+    if (!builtInModules) {
+        builtInModules = computeBuiltInModules();
+    }
+
+    return builtInModules;
+
+    async function computeBuiltInModules() {
+        // These modules are built-in to Node.js, and are available via `require(...)`
+        // but are not stored in the `require.cache`.  They are guaranteed to be
+        // available at the unqualified names listed below. _Note_: This list is derived
+        // based on Node.js 6.x tree at: https://github.com/nodejs/node/tree/v6.x/lib
```

Comment:
Should we update this to the v10 or v11 tree?

Reply:
potentially.... i'm willing to wait on customer requests :)